File: Modules/SEDO/EMS/SIE.Web.EMS/SpareParts/Applys/Commands/DelAppCommand.js
SIE.defineCommand('SIE.Web.EMS.SpareParts.Applys.Commands.DelAppCommand', {
    extend: 'SIE.cmd.Delete',
    meta: { text: "删除", group: "edit", iconCls: "icon-DeleteEntity icon-red" },

    canExecute: function (view) {
        if (view.getSelection() == null || view.getSelection().length == 0) {
            return false;
        }

        var state = view.getCurrent().getAuditState();
        if (state != 0 && state!=2) {
            return false;
        }
        return true;
    },

    execute: function (view, source) {
        var sel = view.getSelection();
        SIE.Msg.askQuestion(Ext.String.format('你确定删除选择的{0}条数据吗？'.t(), sel.length), function () {
            view.execute({
                data: view.getSelectionIds(),
                success: function (res) {
                    var count = res.Result;
                    SIE.Msg.showMessage(Ext.String.format("成功删除{0}条数据。".L10N(),count));
                    view.reloadData();
                },
            });
        });
    }
});

File: Modules/SEDO/EMS/SIE.Web.EMS/SpareParts/Applys/Commands/DelAppCommand.cs
using SIE.EMS.SpareParts.Applys.Controllers;
using SIE.Web.Command;
using System.Collections.Generic;

namespace SIE.Web.EMS.SpareParts.Applys.Commands
{
    /// <summary>
    /// 删除
    /// </summary>
    [JsCommand("SIE.Web.EMS.SpareParts.Applys.Commands.DelAppCommand")]
    public class DelAppCommand : ViewCommand
    {
        /// <summary>
        /// 执行
        /// </summary>
        /// <param name="args"></param>
        /// <param name="scope"></param>
        /// <returns></returns>
        protected override object Excute(ViewArgs args, string scope)
        {
            List<double> ids = args.Data.ToJsonObject<List<double>>();

            int count = RT.Service.Resolve<SparePartAppController>().DeleteApp(ids);
            return count;
        }
    }
}
